src/WebJobs.Script.WebHost/Security/DefaultSecretsRepositoryFactory.cs

```diff
@@ -19,8 +19,10 @@ public ISecretsRepository Create(ScriptSettingsManager settingsManager, WebHostS
             string storageString = AmbientConnectionStringProvider.Instance.GetConnectionString(ConnectionStringNames.Storage);
             if (secretStorageType != null && secretStorageType.Equals("Blob", StringComparison.OrdinalIgnoreCase) && storageString != null)
             {
-                string siteHostId = settingsManager.AzureWebsiteDefaultSubdomain ?? config.HostConfig.HostId;
-                return new BlobStorageSecretsRepository(Path.Combine(webHostSettings.SecretsPath, "Sentinels"), storageString, siteHostId);
+                string siteNameId = settingsManager.GetSetting(EnvironmentSettingNames.AzureWebsiteName) ?? config.HostConfig.HostId;
+                string slotName = settingsManager.GetSetting(EnvironmentSettingNames.AzureWebsiteSlotName);
+                string siteSlotName = string.IsNullOrEmpty(slotName) ? siteNameId : string.Format("{0}-{1}", siteNameId, slotName);
```

Comment:
A little concerned about what this logic opens up. If we were ever in a position where the slot name wasn't initially available and was eventually set, we'd end up looking for secrets in the wrong location and breaking the app. We need to make sure this is stable.

Is the primary/production slot always called `production`? If so, we should treat `production` and not set as the same and not use the slot name in that case.